improving adding a kit docs (#81)
MDAnalysis/MDAKits: docs/source/add.rst

     with fixing your submission.
   * If the tests passed, add a comment to your PR with the text

        *@MDAnalysis/mdakits-reviewers, ready for review.*

#. Someone from the MDAnalysis team will *review* your Pull Request for
   fulfilling the :ref:`requirements for an MDAKit<requirements>` and for any
   technical issues.

   They may ask for modifications so please *check your PR regularly*
   or `enable GitHub notifications`_. Part of the registration process
   is to engage in a conversation with the MDAnalysis team. The PR
   cannot be merged until all relevant questions have been addressed.
#. When your PR passes review, it will be merged and your MDAKit is registered.
#. Remember: **You remain responsible for maintaining your code**. The
   registry (and MDAnalysis) continuously test your code but the
   MDAnalysis team cannot fix it for you.

   .. Note:: 

      The registry will alert you via GitHub issues when your tests
      start failing with newer versions of MDAnalysis or other
      dependencies. In this case **please fix your code and make a new
      release of your code**.  

.. _specification:
   

> Specification of the ``metadata.yaml`` file

Author: @IAlibay I added this preliminary specification section — mainly so that we can link to the anchor and update later. 

I found that I was piling specs into the "template" section that are actually better described stand-alone. Eventually we can document all entries in this section independent from the template but let's wait until the dust has settled. For right now, I am happy with using the template/metadata.yaml as ground truth.

===========================================

The ``metadata.yaml`` file is in `YAML format`_.

The file contains required and optional entries.

**Required entries**
   Any variable in the *Required entries* section **must** be
   provided. If missing or empty, the MDAKit cannot be registered.

**Optional entries**
   Any variable in the *Optional entries* section **may** be left
   empty or be left out. However, it is recommended to provide as much

>    information as possible.

Reviewer: ```suggestion
   information as possible. Some aspects of the registry checks may not work if these details are missing.
```

Small typo here -- just modifying @IAlibay's suggestion

.. Note::    

   For now, the :ref:`template file<template>` serves as the primary
   specification for the MDAKits registry ``metadata.yaml`` file.  The
   MDAKits registry specification will likely be updated in the future

>    and some of the optional entries may become required.

Reviewer 2: ```suggestion
   with new optional and required entries. The MDAKits team will assist developers with migrations should they become necessary.
```

Do you have an MDAKit? Consider adding it to the `MDAKit registry`_!

.. note::   

>    The `MDAKit registry`_ is still in its infancy, we expect that the way in

Reviewer: ```suggestion
   The `MDAKit registry`_ is still in its initial stages. We expect that the way in
```

   over time. Please reach out via the `issue tracker`_ if you have any
   questions.

Registering an MDAKit requires you to create a single file with metadata
(called ``metadata.yaml``). You add this file to the `MDAnalysis/mdakits
repository`_ on GitHub.

Process
=======

Add a MDAKit by following these steps:

1. Create a fork the MDAKit repository https://github.com/MDAnalysis/mdakits.
2. Make a new branch (choose any name, but "add-my-awesome-mdakit" is a good
   one).
3. Add a new folder under ``mdakits`` with the name of your MDAKit.
4. Add a metadata YAML file with your MDAKit's details; copy the template from
   `mdakits/template/metadata.yaml`_ and modify it. See the comments in the
   file and the :ref:`notes below<template>` for more explanations.
5. Open a Pull Request (PR) against the *main* branch of the
   `MDAnalysis/mdakits repository`_.
6. Someone from the MDAnalysis team will *review* your Pull Request for
   fulfilling the :ref:`requirements for an MDAKit<requirements>` and for any
   technical issues. They may ask for modifications so please *check your PR
   regularly* or `enable GitHub notifications`_. Part of the registration

>    process is to engage in a conversation with the MDAnalysis team so please be

Reviewer 2: I don't think we need to go this far as to ask folks to be responsive. I think it's ok if we just say that it's a conversation and it won't be merged until all the relevant questions have been answered?

Author: changed

   responsive.
7. When your PR passes review, it will be merged and your MDAKit is registered.

> 

Reviewer 2: We should add a 8. here where we mention about checks, failures, and updates - i.e. it is their responsibility to update things as necessary.

Author: added ... please have a look in a moment

.. _template:

Template ``metadata.yaml`` file
===============================

The following is the template metadata file
`mdakits/template/metadata.yaml`_. Please see the *comments in the template*
for an explanation of all possible items. You may delete or rewrite any
comments. Replace values with content appropriate for your MDAKit.

.. Note::

   The template uses the strings ``GH_HOST_ACCOUNT``,  ``MYPROJECT``, and
   ``MYPACKAGE`` to indicate that you should be filling in *your* details.

   
``GH_ACCOUNT``
   The GitHub account where he source code repository is held, typically your

>    username or the organization that you're part off.

Reviewer: ```suggestion
   username or the organization that you're part of.
```

``MYPROJECT``

>    The name of your project. This is name of your repository. In the template we

Reviewer 2: ```suggestion
   The name of your project. This is the name of your repository. In the template we
```

>    assume that this is also the PyPi/conda package name but you can make them

Reviewer 2: This needs checking, I'm not sure it's true.

Author: ok, rephrased 

   The name of your project. This is the name of your repository. In the template we
   also use it as the PyPi/conda package name.

Reviewer 2: So I double checked here and you're right - the conda/pip install is freeform so `package_name` doesn't get injected.

The only place where `package_name` is a hard requirement is for codecov, it's a piece of code that currently doesn't work but eventually we'll need up and running (for the badges).

Author: Then let's leave it as the more restrictive phrasing for right now.

   different.
   
``MYPACKAGE``

>    The name of the Python package, it describes is how you import it in Python

Reviewer: ```suggestion
   The name of the Python package. It describes how you import it in Python
```

>    code.

Reviewer: ```suggestion
   code, i.e. it is used in ``import MYPACKAGE``.
```

   maintainers:
     - NAME1
     - OPTIONAL_NAME2
     - OPTIONAL_NAME3
       
   ## str: a free form description of the mdakit
   description:
       (REPLACE WITH A SHORT DESCRIPTION OF WHAT YOUR MDAKit DOES.)
       
   ## List(str): a list of keywords which describe the mdakit
   keywords:
     - KEYWORD1
     - KEYWORD2
       
   ## str: the license the mdakit falls under
   license: GPL-2.0-or-later
   
   ## str: the link to the project's code
   project_home: https://github.com/GH_HOST_ACCOUNT/MYPROJECT/
   
   ## str: the link to the project's documentation
   documentation_home: https://MYPROJECT.readthedocs.io
   
   ## str: the type of documentation available [UserGuide, API, README]
   documentation_type: UserGuide + API

   #------------------------------------------------------------
   # Optional entries
   #------------------------------------------------------------   
   ## List(str): a list of commands to use when installing the latest
   ## release of the code. Note: only one installation method can currently
   ## be defined. We suggest using mamba where possible (e.g.
   ##   mamba -c conda-forge install MYPROJECT
   ## for a conda package installation).

>    ## Here we use the simple PyPi installation:

Reviewer 2: ```suggestion
   ## Here we use a simple PyPi installation:
```

   install:
     - pip install MYPROJECT
       
   ## List(str): a list of commands to use when installing the mdakit from its
   ## source code.
   src_install:
     - pip install git+https://github.com/GH_HOST_ACCOUNT/MYPROJECT@main
       
   ## str: the package name used to import the mdakit
   import_name: MYPACKAGE
   
   ## str: a specification for the range of Python versions supported by this MDAKit
   python_requires: ">=3.9"
   
   ## str: a specification for the range of MDAnalysis versions supported by this MDAKit
   mdanalysis_requires: ">=2.0.0"
   
   ## List(str): a list of commands to use when attempting to run the MDAKit's tests
   ## If you package your tests inside your package then you can typically use the 
   ##     pytest --pyargs MYPACKAGE.tests
   ## command as shown below. 
   ## Otherwise you need to include commands to make the tests available. 
   ## For example, if the tests are in the repository at the top level under `./tests`:

>    ## First use `git clone latest` to either clone the top commit for "develop" runs or check out

Reviewer 2: ```suggestion
   ## First use `git clone latest` to either clone the top commit for "development code" checks or check out
```

   ## the latest tag for "latest release" checks. Then then run pytest:
   ##    - git clone latest
   ##    - pytest -v ./tests
   ## Feel free to ask for advice on your pull request!
   run_tests:

>      - pytest --pyargs MYPACKAGE.tests

Reviewer 2: You shouldn't need to have the `.tests` here? I need to double check but package test discovery should work.

Author: ok, removing .tests

Reviewer 2: Can confirm the auto discovery works using our favourite new mdakit (`solvation_analysis`).

       
   ## List(str): a list of commands to use to install the necessary dependencies required
   ## to run the MDAKit's tests.
   ## The default below _might_ be sufficient or you might not even need MDAnalysisTests:
   ## make sure that it is appropriate for how you run tests.
   test_dependencies:
     - mamba install pytest MDAnalysisTests
       
   ## str: the organisation name the MDAKit falls under
   project_org: GH_HOST_ACCOUNT
   
   ## str: the development status of the MDAKit

>    ## See https://pypi.org/classifiers/ for development status classifiers.

Reviewer 2: We should do the same for the license classifier.

Author: The example    
```yaml
license: GPL-2.0-or-later
```
does not agree with 
```
License :: OSI Approved :: GNU General Public License v2 or later (GPLv2+)
```
from https://pypi.org/classifiers/

Given that _we_ do not seem to be following the trove classifiers I won't add a specific link. That seems to be an "raise an issue to decide what we want to use".

Reviewer 2: I think we're using the conda-forge definition of licenses, let me dig that up.

Reviewer 2: Here you go: https://spdx.org/licenses/

That's what we should be using in my opinion. It's concise and (assuming I ever get the time) we should be able to do a pydantic validator that checks against that the spdx list.

Author: thanks, added

>    development_status: Productions/Stable

Reviewer: ```suggestion
   development_status: Production/Stable
```

   
   ## List(str) a list of publications to cite when using the MDAKit
   ## Links to scientific publications or stable URLs (typically of the form
   ## https://doi.org/<DOI> or to a preprint server)
   publications:
     - URL1
     - URL2
       
   ## str: a link to the MDAKit's community (mailing list, github discussions, etc...)
   community_home: URL
   
   ## str: a link to the MDAKit's changelog

>    changelog: https://github.com/MYNAME/MYPROJECT/blob/main/CHANGES

Reviewer: ```suggestion
   changelog: https://github.com/MYNAME/MYPROJECT/blob/main/CHANGELOG.md
```

That is what is used in the cookiecutter
